plugins/analysis-phonenumber/src/main/java/org/opensearch/analysis/phone/PhoneNumberAnalysisPlugin.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.analysis.phone;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.opensearch.index.analysis.AnalyzerProvider;
+import org.opensearch.index.analysis.TokenizerFactory;
+import org.opensearch.indices.analysis.AnalysisModule;
+import org.opensearch.plugins.AnalysisPlugin;
+import org.opensearch.plugins.Plugin;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * This plugin provides an analyzer and tokenizer for fields which contain phone numbers, supporting a variety of formats
+ * (with/without international calling code, different country formats, etc.).
+ */
+public class PhoneNumberAnalysisPlugin extends Plugin implements AnalysisPlugin {
+
+    @Override
+    public Map<String, AnalysisModule.AnalysisProvider<AnalyzerProvider<? extends Analyzer>>> getAnalyzers() {
```

Comment:
This is just a question: do you think we will be better off with `phone-ngrams` and `phone` names for analyzers?  The `phone-search` is somewhat confusing (at least to me) since in both cases the phone could be searched, right?

Reply:
i followed the pattern of the old ES plugin (but nobody is using that anymore, so no preservation of vested rights needed there).
the reasoning is this:
you really, really should use the `phone` analyzer as the normal analyzer and `phone-search` as the search analyzer. deviating from this doesn't make a lot of sense as using the ngram analyzer (which is the normal one, not the search one!) on the search input would make the search input match everything which begins the same way as the phone number, which isn't what you want

Reply:
Got it, thank you for explaining